hcl/hcl.go

```diff
@@ -597,42 +597,38 @@ func parseRootConfig(cfg *RootConfig, block *hclsyntax.Block) error {
 		)
 	}
 
-	foundGit := false
-	foundGenerate := false
-
 	logger.Trace().Msg("Range over blocks.")
 
 	for _, b := range block.Body.Blocks {
 		switch b.Type {
 		case "git":
 			logger.Trace().Msg("Type was 'git'.")
 
-			if foundGit {
+			if cfg.Git != nil {
 				return errutil.Chain(
 					ErrMalformedTerramateConfig,
 					fmt.Errorf("multiple terramate.config.git blocks"),
 				)
 			}
 
-			foundGit = true
-			logger.Trace().
-				Msg("Parse git config.")
-			err := parseGitConfig(&cfg.Git, b)
+			cfg.Git = &GitConfig{}
+
+			logger.Trace().Msg("Parse git config.")
+
+			err := parseGitConfig(cfg.Git, b)
 			if err != nil {
 				return err
 			}
 		case "generate":
 			logger.Trace().Msg("Found block generate")
 
-			if foundGenerate {
+			if cfg.Generate != nil {
 				return errutil.Chain(
 					ErrMalformedTerramateConfig,
-					fmt.Errorf("multiple terramate.config.generate blocks"),
+					fmt.Errorf("multiple terramate.config.backend blocks"),
```

Comment:
we need to provide more details about which files have the duplicated entry, but we can do it in a separate PR.

Reply:
yeah, right now the files from which each block came from are not handy, would require some changes on the internal design (but indeed we need to improve that)